src/components/controller.cpp

```diff
@@ -386,36 +390,195 @@ void controller::process_eventqueue() {
+#endif
+    A_MAP("xbacklight+", xbacklight_module, EVENT_INC),
+    A_MAP("xbacklight-", xbacklight_module, EVENT_DEC),
+    A_MAP("backlight+", backlight_module, EVENT_INC),
+    A_MAP("backlight-", backlight_module, EVENT_DEC),
+#if ENABLE_XKEYBOARD
+    A_MAP("xkeyboard/switch", xkeyboard_module, EVENT_SWITCH),
+#endif
+#if ENABLE_MPD
+    A_MAP("mpdplay", mpd_module, EVENT_PLAY),
+    A_MAP("mpdpause", mpd_module, EVENT_PAUSE),
+    A_MAP("mpdstop", mpd_module, EVENT_STOP),
+    A_MAP("mpdprev", mpd_module, EVENT_PREV),
+    A_MAP("mpdnext", mpd_module, EVENT_NEXT),
+    A_MAP("mpdrepeat", mpd_module, EVENT_REPEAT),
+    A_MAP("mpdsingle", mpd_module, EVENT_SINGLE),
+    A_MAP("mpdrandom", mpd_module, EVENT_RANDOM),
+    A_MAP("mpdconsume", mpd_module, EVENT_CONSUME),
+    // Has data
+    A_MAP("mpdseek", mpd_module, EVENT_SEEK),
+#endif
+    // Has data
+    A_MAP("xworkspaces-focus=", xworkspaces_module, EVENT_FOCUS),
+    A_MAP("xworkspaces-next", xworkspaces_module, EVENT_NEXT),
+    A_MAP("xworkspaces-prev", xworkspaces_module, EVENT_PREV),
+    // Has data
+    A_MAP("bspwm-deskfocus", bspwm_module, EVENT_FOCUS),
+    A_MAP("bspwm-desknext", bspwm_module, EVENT_NEXT),
+    A_MAP("bspwm-deskprev", bspwm_module, EVENT_PREV),
+#if ENABLE_I3
+    // Has data
+    A_MAP("i3wm-wsfocus-", i3_module, EVENT_FOCUS),
+    A_MAP("i3wm-wsnext", i3_module, EVENT_NEXT),
+    A_MAP("i3wm-wsprev", i3_module, EVENT_PREV),
+#endif
+    // Has data
+    A_MAP("menu-open-", menu_module, EVENT_OPEN),
+    A_MAP("menu-close", menu_module, EVENT_CLOSE),
+  };
+#undef A_MAP
+  // clang-format on
+
+  // Check if any key in the map is a prefix for the `cmd`
+  for (const auto& entry : legacy_actions) {
+    const auto& key = entry.first;
+    if (cmd.compare(0, key.length(), key) == 0) {
+      string type = entry.second.first;
+      auto data = cmd.substr(key.length());
+      string action = entry.second.second;
+
+      // Search for the first module that matches the type for this legacy action
+      for (auto&& mod : m_modules) {
+        if (mod->type() == type) {
+          auto handler_ptr = std::dynamic_pointer_cast<input_handler>(mod);
+          auto handler_name = handler_ptr->input_handler_name();
+          // TODO make this message more descriptive and maybe link to some documentation
+          // TODO use route to string methods to print action name that should be used.
+          if (data.empty()) {
+            m_log.warn("The action '%s' is deprecated, use '#%s.%s' instead!", cmd, handler_name, action);
+          } else {
+            m_log.warn("The action '%s' is deprecated, use '#%s.%s.%s' instead!", cmd, handler_name, action, data);
+          }
+          m_log.info(
+              "Forwarding legacy action '%s' to module '%s' as '%s' with data '%s'", cmd, handler_name, action, data);
+          if (!handler_ptr->input(std::forward<string>(action), std::forward<string>(data))) {
```

Comment:
Maybe we should pass these parameters as const ref since we always need to copy them (line 471 and 473)